src/main/java/teammates/common/datatransfer/attributes/AccountRequestAttributes.java

```diff
@@ -16,41 +17,105 @@
  */
 public final class AccountRequestAttributes extends EntityAttributes<AccountRequest> {
 
-    private String email;
     private String name;
+    private String pureInstitute;
+    private String pureCountry;
     private String institute;
-    private Instant registeredAt;
+    private String email;
+    private String homePageUrl;
+    private String comments;
+    private AccountRequestStatus status;
     private Instant createdAt;
+    private Instant lastProcessedAt;
+    private Instant registeredAt;
     private transient String registrationKey;
 
-    private AccountRequestAttributes(String email, String institute, String name) {
-        this.email = email;
-        this.institute = institute;
+    private AccountRequestAttributes(String name, String pureInstitute, String pureCountry, String institute,
+                                     String email, String homePageUrl, String comments) {
         this.name = name;
+        this.pureInstitute = pureInstitute;
+        this.pureCountry = pureCountry;
```

Comment:
These two fileds are necessary here in order to validate institute and country separately. The main reason to validate them separately is because I want to hide the details of generating the final institute to end users. Also only this design allows the country field to be a drop-down in the future. If I understand data transfer classes correctly, they are used to facilitate transferring data among componenets instead of simply being a mirror of the corresponding entity class. And this `AccountRequestAttributes` class was very carefully designed for this purpose. This also replies to your next few comments.

Reply:
There is no need for the BE to know about country and institute.

> Also only this design allows the country field to be a drop-down in the future. 

This is not affected

> These two fileds are necessary here in order to validate institute and country separately.

neither is this

> they are used to facilitate transferring data among componenets instead of simply being a mirror of the corresponding entity class.

Yes, not to validate fields that are not used in the application...

Reply:
How would you design it to be a drop-down list? It needs all the countries to be stored as a list in the backend, and an API to return the list to FE. Later, it validates it by checking it is one of the entity in the list.

Reply:
This is simply not true. I do not have the capacity to explain every little detail to you. Please trust me when I say this design is extremely messy and unnecessarily complex.

Reply:
Nope unless you'll prove it to me. I also know it's very very complex but I don't think it's unnecessary. It took me quite long to design and test the AccountRequestAttributes, considering all the involving places. It also affects a lot of places and can take a few days to change. The current design is the best to me.

Reply:
I also know it's very easy to follow other classes and simply combine it in the FE. However, that won't be a proper way of doing things here because for users, they are two fields.

Reply:
@samuelfangjw I am considering an alternative way to manually call FieldValidator to validate those fields, then join country and institute together in the `CreateAccountRequestAction`. The **potential** drawbacks are it won't utilise `getInvalidityInfo` and validation is done inside `CreateAccountRequestAction`. The cons it brings are `pureInstitute` and `pureCountry` can be removed from `AccountRequestAttributes`, and also the current prefixes won't be needed.

Reply:
I deem this is a better way than the current and I will adopt it first.

Reply:
Anyway the country definitely needs to be passed to the backend instead of being joined in frontend. The only problem is where to join it.